middleware/queue/source/forward/handle.cpp

```diff
@@ -575,8 +577,15 @@ namespace casual
                            if( message.state == decltype( message.state)::busy)
                               return;
 
+                           // if we cant find a pending lookup we assume the lookup is discarded and handle it later
+                           auto found = common::algorithm::find( state.pending.service.lookups, message.correlation);
+                           if( ! found)
+                           {
+                              log::line( verbose::log, "no pending lookup found for correlation: ", message.correlation);
+                              return;
+                           }
```

Comment:
When we reach this, there should be a pending _lookup::discard_, should we check this and log error if not? 
```
if( ! algorithm::find( state.pending.service.discard_lookup, message.correlation)
    log::line( log::category::error, code::casual::invalid_semantics, " expected descard_lookup state for correlation: ", message.correlation);
```
It sure would help us, if for some reason this state is reached. 
Or, maybe we can rely on the discard reply any way... Hmm, for me it feels like a good thing that every _handler_ sort of taker care of the expected state. In this case, if it's not in pending lookups, it should be in pending discard, if so, I'm good. But my thinking might be overkill..